cmd/evm/blockrunner.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/tests"
+
+	"github.com/urfave/cli/v2"
+)
+
+var blockTestCommand = &cli.Command{
+	Action:    blockTestCmd,
+	Name:      "blocktest",
+	Usage:     "executes the given blockchain tests",
+	ArgsUsage: "<file>",
+}
+
+// BlocktestResult contains the execution status after running a blockchain test, any
+// error that might have occurred and a dump of the final blockchain if requested.
+type BlocktestResult struct {
+	Name  string      `json:"name"`
+	Pass  bool        `json:"pass"`
+	Fork  string      `json:"fork"`
+	Error string      `json:"error,omitempty"`
+	State *state.Dump `json:"state,omitempty"`
+}
```

Comment:
I suppose the intention was to actually use this? 
Right now, it's totally silent if the test passes
```
$ ./evm blocktest /home/user/go/src/github.com/ethereum/go-ethereum/tests/testdata/BlockchainTests/GeneralStateTests/stChainId/chainId.json 
$
```